### PR TITLE
Implement the /bootstrap control plane API endpoint

### PR DESCRIPTION
**Pull request by freeekanayaka**
This branch adds a new "/bootstrap" endpoint to the internal control
plane API for the fake-jujud process.

It can be used to perform a fake "bootstrap" of a controller bound to
the dummy provider.

This will happen in the "suite.SetUpTest(c)" call which is basically
what the HTTP API triggers (under the hood the juju testing machinery
will setup the controller).

The branch also adds a minimal HTTP client for talking to the control
plane API and hitting the "/bootstrap" endpoint.

A test has been added to demonstrate that the client and server work
end-to-end. More testing will be added in follow-up branches to assert
that a working controller gets actually greated.

### CODE REVIEW

File: common/core/github.com/juju/juju/api/client-fakejuju.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	response, err := http.Post(url, "", bytes.NewBuffer([]byte("")))
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	logger.Debugf("Got response Status: %s", response.Status)
+
+	if response.StatusCode != 200 {
+		return errors.New("Failed fake-juju request")
```
Comment (sparkiegeek) [inline common/core/github.com/juju/juju/api/client-fakejuju.go]: it'd be nice if this included what error it got back...

File: common/service/api.go [hunk 2]
```diff
@@ -42,3 +43,22 @@ func (f *FakeJujuRunner) stopControlPlaneAPI() {
 	log.Infof("Stopping control plane API on address %s", addr.String())
 	f.listener.Close()
 }
+
+// Bootstrap the controller by starting machine 0
+func (f *FakeJujuRunner) bootstrap(w http.ResponseWriter, req *http.Request) {
+	command := newCommand(commandCodeBootstrap)
+	f.commands <- command
+	writeResponse(w, <-command.done)
```
Comment (sparkiegeek) [inline common/service/api.go]: missing space? hmm does Travis run go fmt ?

Comment (freeekanayaka) [inline common/service/api.go]: Travis doesn't run go fmt, but this is go fmt'ed (so yes, no space).

File: common/core/github.com/juju/juju/api/client-fakejuju.go (hunk 1 above)
Comment (freeekanayaka) [inline common/core/github.com/juju/juju/api/client-fakejuju.go]: Fixed.

**Pull request merged**